src/Ast.ml

```diff
@@ -225,6 +225,12 @@ and Requires_sub_terms : sig
 
   val exposed : cls -> expression -> bool
 
+  type ltgt = LT | GT
+
+  val typ_exposed_left : ltgt -> core_type -> bool
+
+  val typ_exposed_right : ltgt -> core_type -> bool
```

Comment:
For my understanding, is it the case that currently there is some redundancy since exposed_left is always concerned with LT and ignores GT, and similarly exposed_right is always concerned with GT and ignores LT? Do you have a generalization in mind that will be needed, or would it make sense to add the most specialized version for now and generalize later when it becomes clear what will be needed?